python_modules/dagit/dagit/webapp/src/Config.tsx

```diff
@@ -5,85 +5,91 @@ import { UL, Colors } from "@blueprintjs/core";
 import TypeWithTooltip from "./TypeWithTooltip";
 import Description from "./Description";
 import { ConfigFragment } from "./types/ConfigFragment";
+import { ConfigPipelineTypesFragment } from "./types/ConfigPipelineTypesFragment";
 
 interface ConfigProps {
   config: ConfigFragment;
+  pipeline: ConfigPipelineTypesFragment;
 }
 
 export default class Config extends React.Component<ConfigProps, {}> {
   static fragments = {
     ConfigFragment: gql`
       fragment ConfigFragment on Config {
         type {
+          ...TypeWithTooltipFragment
           __typename
           name
-          description
+        }
+      }
+      ${TypeWithTooltip.fragments.TypeWithTooltipFragment}
+    `,
+    ConfigPipelineTypesFragment: gql`
+      fragment ConfigPipelineTypesFragment on Pipeline {
+        types {
+          __typename
           ... on CompositeType {
             fields {
               name
               description
               isOptional
               defaultValue
               type {
+                __typename
                 name
-                description
-                ...TypeWithTooltipFragment
-                ... on CompositeType {
-                  fields {
-                    name
-                    description
-                    isOptional
-                    defaultValue
-                    type {
-                      name
-                      description
-                      ...TypeWithTooltipFragment
-                    }
-                  }
-                }
               }
             }
           }
           ...TypeWithTooltipFragment
         }
       }
-
       ${TypeWithTooltip.fragments.TypeWithTooltipFragment}
     `
   };
 
-  renderFields(config: ConfigFragment) {
-    if (config.type.__typename === "CompositeType") {
-      return (
-        <UL>
-          {config.type.fields.map((field, i: number) => (
-            <li key={i}>
-              {field.name} {field.isOptional ? "(optional)" : null}{" "}
-              <TypeWithTooltip type={field.type} />
-              <DescriptionWrapper>
-                <Description description={field.description} />
-              </DescriptionWrapper>
-            </li>
-          ))}
-        </UL>
-      );
-    } else {
-      return null;
+  public render() {
+    const {
+      pipeline: { types },
+      config
+    } = this.props;
+
+    let result = "{";
```

Comment:
this is a very unittest-able thing. let's get in the habit of writing jest tests for this type of thing